src/x11/background_manager.cpp

```diff
@@ -189,7 +189,7 @@ void bg_slice::allocate_resources(const logger& log, xcb_visualtype_t* visual) {
 }
 
 void bg_slice::free_resources() {
-  m_surface.release();
```

Comment:
Release doesn't free. This function releases the ownership of the managed object. So there is a memory leak.
`smart_ptr.reset()` or `smart_ptr = nullptr` must be used to delete the object.